ocp_utilities/data_collector.py

```diff
@@ -0,0 +1,165 @@
+        base_directory=directory,
+    )
+
+    virt_launcher_pod = vmi.virt_launcher_pod
+    write_to_file(
+        file_name=f"{virt_launcher_pod.name}.log",
+        content=virt_launcher_pod.log(container="compute"),
+        base_directory=directory,
+    )
+    write_to_file(
+        file_name=f"{virt_launcher_pod.name}.yaml",
+        content=virt_launcher_pod.instance.to_str(),
+        base_directory=directory,
+    )
+
+
+def collect_data_volume_data(dyn_client, directory=None):
+    cdi_pod_prefixes = ("importer", "cdi-upload")
+    for pod in Pod.get(dyn_client=dyn_client):
+        pod_name = pod.name
+        if pod_name.startswith(cdi_pod_prefixes) or pod_name.endswith("source-pod"):
+            write_to_file(
+                file_name=f"{pod_name}.log", content=pod.log(), base_directory=directory
+            )
+            write_to_file(
+                file_name=f"{pod_name}.yaml",
+                content=pod.instance.to_str(),
+                base_directory=directory,
+            )
+
+
+def collect_data(directory, resource_object, collect_pod_logs):
+    LOGGER.info(f"Collecting instance data for {resource_object.name}")
+
+    if resource_object.kind == ProjectRequest.kind:
+        resource_object = Namespace(name=resource_object.name)
+
+    write_to_file(
+        file_name=f"{resource_object.name}.yaml",
+        content=resource_object.instance.to_str(),
+        base_directory=directory,
+    )
+    if resource_object.kind == VirtualMachine.kind and resource_object.vmi:
+        collect_vmi_data(vmi=resource_object.vmi, directory=directory)
+
+    if resource_object.kind == DataVolume.kind:
+        collect_data_volume_data(dyn_client=resource_object.client, directory=directory)
+
+    if resource_object.kind == Pod.kind and collect_pod_logs:
+        container_name = [
+            container["name"]
+            for container in resource_object.instance.status.containerStatuses
+        ][0]
+        write_to_file(
+            file_name=f"{resource_object.name}.log",
+            content=resource_object.log(**{"container": container_name}),
+            extra_dir_name=resource_object.kind,
+            base_directory=directory,
+        )
+
+
+def prepare_test_data_dir(item, prefix, logs_path=None):
+    test_cls_name = item.cls.__name__ if item.cls else ""
+    test_dir_log = os.path.join(
+        logs_path or py_config["data_collector"]["data_collector_base_directory"],
+        item.fspath.dirname.split("/tests/")[-1],
+        item.fspath.basename.partition(".py")[0],
+        test_cls_name,
+        item.name,
+        prefix,
+    )
+    os.makedirs(test_dir_log, exist_ok=True)
+    return test_dir_log
+
+
+def collect_resources_yaml_instance(resources_to_collect, namespace_name=None):
+    get_kwargs = {"dyn_client": infra.get_admin_client()}
+    for _resources in resources_to_collect:
+        if _resources == Service:
+            get_kwargs["namespace"] = namespace_name
```

Comment:
This would stop you from being able to collect any non-namespaced resource/ resource from other namespace, unless `get_kwargs = {"dyn_client": infra.get_admin_client()}` is moved inside for loop.

Reply:
I am not following what you mean; can you please explain why we need to move it inside the loop

Reply:
get_kwargs is set outside the for loop. if resources_to_collect contains [a, b, Service, d, e], for d and e get_kwargs would be {"dyn_client": infra.get_admin_client(), "namespace": namespace_name}. So if d and e does not belong in namespace_name, those won't be collected.